src/Cello/problem_Problem.cpp

```diff
@@ -444,6 +446,8 @@ void Problem::initialize_method ( Config * config ) throw()
   const size_t num_method = config->method_list.size();
 
   Method::courant_global = config->method_courant_global;
+
+  method_list_.push_back(new MethodNull(config->method_null_dt)); 
```

Comment:
This addition (coupled with the change in `MethodNull` to refresh all fields) creates a large problem for the VL+CT solver (in PR #9 ).

Presently, issues arise when refresh is scheduled to refresh all fields and there are registered temporary fields that have been deallocated. This is an issue for the VL+CT solver because the current implementation of its `compute` method presently allocates and deallocates a series of temporary fields during every evaluation (of non-leaf blocks). For a host of reasons, I have plans to move away from this approach, but I'm not sure I'm going to be able to get to that immediately.

Is there way we can work around this in the short-term? Maybe we can add an instance of `MethodNull` to `method_list_` after the for-loop if `method_list_` contains no entries? (If you could explain why you added this line, I might be able to make more constructive suggestions).

Reply:
How is the problem manifesting itself? The purpose of inserting MethodNull at the beginning of the list of methods is (in part) to ensure all fields are refreshed after the adapt phase.  It should only affect leaf blocks, I don't know why you're having problems with non-leaf blocks. Moving the refresh step to the end of the adapt phase would be cleaner, but functionally identical, so that wouldn't solve anything. Would skipping temporary fields when using add_all_fields() solve the conflict?

Reply:
Apologies for the delayed response.

The problem manifests during refresh in two different ways based on the centering of the first temporary field that the refresh machinery handles:
1. If the first handled temporary field has `cx`,`cy`,`cz` values that are all 0 or 1, a segmentation fault occurs (because the refresh machinery is trying to access a `NULL` pointer). The traceback says this occurs in `FieldFace::face_to_face`
2. If any of the `cx`,`cy`,`cz` values are `-1`, than it triggers an error (in `FieldFace::face_to_face`'s call to `FieldFace::loop_limits`) that I previously added to the master branch that informs the user that refreshing this type of face-centered field is not currently supported.

I don't think that the leaf/non-leaf blocks have anything to do with this problem since I have not currently not tested AMR. I think I may have accidentally obfuscated the issue by mentioning this at all.

I think skipping temporary fields when using `add_all_fields()` would solve the problem. Would this actually involve changing `FieldFace::field_list_src_` and `FieldFace::field_list_dest_`.

If you want to replicate this problem on this branch, just replace the implementation of `MethodNull::compute` with:

```c++
void MethodNull::compute( Block * block) throw(){
  std::vector<int> id_l;
  std::vector<std::string> names = {"temp0", "temp1", "temp2"};
  std::vector<int> cz_l = {0,-1,1};
  Field field = block->data()->field();
  FieldDescr * field_descr = field.field_descr();

  // first reserve and allocate temporary fields:
  if (field.is_field(names[0])){
    // the temporary field was added in a previous time-step
    for (std::string name : names){ id_l.push_back(field.field_id(name)); }
  } else {
    // Reserve a temporary field
    for (std::size_t i=0; i<names.size(); i++){
      id_l.push_back(field_descr->insert_temporary(names[i]));
      field_descr->set_centering(id_l[i],0,0,cz_l[i]);
    }
  }
  for (int id : id_l){ field.allocate_temporary(id); }

  // --- do some work ---

  // deallocate the temporary field:
  for(int id : id_l){ field.deallocate_temporary(id); }

  block->compute_done();
}
```

Reply:
Skipping temporary fields when using `add_all_fields()` appears to solve the problem. I had modified `FieldFace::field_list_src_` and `FieldFace::field_list_dest_` to only add fields (in the for-loop) if they are permanent. My VL+CT tests now pass.

I'm now verifying that this doesn't cause the rest of the test-suite to break